src/randomizer.py

```diff
@@ -1412,6 +1412,57 @@ def get_child(parent: Union[ca.StructRef, ca.UnaryOp]) -> ca.Node:
     ensure(changed)
 
 
+def perm_split_assignment(
+    fn: ca.FuncDef, ast: ca.FileAST, indices: Indices, region: Region, random: Random
+) -> None:
+    """Split assignments of the form a = b . c . d ...; into a = b; a = a . c . d ...;, a = c . d ...; a = b . a;, etc."""
+    cands = []
+    # Look for assignments of the form 'var = binaryOp' (ignores op=)
+    class Visitor(ca.NodeVisitor):
+        def visit_Assignment(self, node: ca.Assignment) -> None:
+            if node.op == '=' and isinstance(node.rvalue, ca.BinaryOp) and region.contains_node(node):
+                cands.append(node)
+
+    Visitor().visit(fn.body)
+    ensure(cands)
+
+    assign = random.choice(cands)
+    var = assign.lvalue
+
+    binops = []
+
+    def collect_binops(node: ca.BinaryOp) -> None:
+        nonlocal binops
```

Comment:
fwiw this can be removed; nonlocal is only needed when you need to assign to an outer variable, not when poking at it using member functions